### ISSUE TITLE
Pipe PKGBUILD diff into a pager (or print individual diffs)

### ISSUE THREAD

**Issue by marier-nico** [Morganamilo/paru#158]
With many packages from the AUR, having all the diffs printed back to back floods the terminal with a bunch of output. It would be convenient to use a pager like `less` to view the diffs when making upgrades. This way, there would be no need for scrolling through the terminal output (with the mouse, I guess!). The [pager](https://crates.io/crates/pager) crate looks like it might be of use here, if that's a good feature idea.

Otherwise it could be nice to print the diff for each package one at a time and wait for user input between each diff.

tl;dr it would be cool to avoid having to scroll back through the terminal to view the earliest diffs (either by using a pager, or by printing a single diff at a time).

I _might_ be able to create a PR for this, but I'm not familiar with the project, so it would probably take a while if I do! (if you like this feature idea, of course)

**Comment by Morganamilo** [comment]
> Otherwise it could be nice to print the diff for each package one at a time and wait for user input between each diff.

This one can be achieved with `export LESS=SRX`.

Printing them all together into one could be done though.

**Comment by marier-nico** [comment]
Interesting, I wasn't aware that this was already possible! Maybe it would be worth putting somewhere in the man page or in the readme?

Also, do you mean printing everything in one single diff? I'm not sure I understand the use-case for this :thinking:

**Comment by Morganamilo** [comment]
> Also, do you mean printing everything in one single diff? I'm not sure I understand the use-case for this thinking

I thought that's what you were asking for.

If they were all in one instance off less then you could just scroll through it all without interruption.

**Comment by marier-nico** [comment]
Ohh yes yes, that's right! Sorry, I misunderstood what you said :sweat_smile: 

But does `export LESS=SRX` make `paru` use `less`? Or does it make `paru` print a single diff and then wait for user input?

**Comment by Morganamilo** [comment]
paru uses git, git uses less. by default git tells less with only page if the input is longer than the terminal. Explicitly setting what less flags you want stops that.

**Comment by marier-nico** [comment]
I understand now, thanks! Well, in this case, I think it would be convenient if it were possible to pipe all the diffs into a single instance of `less`. Though now that I know about the `LESS` env variable, I feel like it would mostly be just a nice-to-have feature rather than a really big improvement :smile:
